examples/capi-quick-start/aws-cluster.yaml

```diff
@@ -24,6 +24,8 @@ spec:
             cni:
               provider: calico
             nfd: {}
+      - name: awsClusterConfig
```

Comment:
👍 I like having it a separate variable, it enables us to only enable the provider specific handler, no need to have docker cod in the codepath for an AWS CluterClass.

Reply:
I'm thinking we can combine the two and have single variable for each provider that combines generic config and provider specific config. The variable will be the same name regardless of provider but have a different schema.